odoo/addons/base/ir/ir_ui_view.py

```diff
@@ -379,6 +379,11 @@ def write(self, vals):
         self.clear_caches()
         return super(View, self).write(self._compute_defaults(vals))
 
+    def unlink(self):
```

Comment:
Shouldn't this be done in the `studio` module instead? How about testing the `MODULE_UNINSTALL_FLAG` context key to use this logic only upon uninstallation.

Is there a compelling argument to say that we should cascade by default when a view is deleted in normal situations? That does not seem like a common task or case, nor an operation that should be done lightly/silently.

Reply:
What you suggest is what I did originally at https://github.com/odoo/enterprise/pull/2338, but after discussing with @rco-odoo and checking out your commit https://github.com/odoo/odoo/commit/e5f7b5673505d8df3951075e486add96de045650 we didn't quite understand why would one want to keep a customization that no longer inherits from a view (since it was supposed to be deleted) and also because the same bug happens with non-studio customizations.

With a non-studio view the problem is the same, upon reinstallation of the module the view will crash upon being accessed.

I get your point, having a 3k line customization deleted silently can be very annoying for the end-user and I'm sure your original patch had a good reasoning behind it, but, even if we don't delete the customization it is useless as it won't be accessible to the user.

Ultimately I think it's best to discuss this directly during the query builder + batch create demo with rco in the upcoming days :p

Reply:
I think it's a reasonable compromise indeed, will do that, thanks.